exporter/prometheusremotewriteexporter/wal.go

```diff
@@ -0,0 +1,247 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusremotewriteexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/tidwall/wal"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+type walConfig struct {
+	Dir             string        `mapstructure:"path"`
```

Comment:
```suggestion
	Dir             string        `mapstructure:"directory"`
```

Reply:
I had it initially as "dir" but it was requested by @rakyll (and it makes sense to me) to keep it as close to Prometheus as we can per https://docs.google.com/document/d/1cCcoFgjDFwU2n823tKuMvrIhzHty4UDyn0IcfUHiyyI/edit#heading=h.mlf37ibqjgov

Reply:
Given that this is a temporary solution and will be replaced by generic storage extension (we agree on that, didn't we? if not can't merge :-) ) I suggest that we make this uniform with storage extension configuration option (and it uses `directory).
I believe we should generally aim for uniformness within the Collector, and only then consider the uniformity with a particular 3rd party technology we support.